pages/applications/view/[id]/[person]/index.tsx

```diff
@@ -157,20 +157,32 @@ export default function ApplicationPersonPage({
                   >
                     Living Situation
                   </HorizontalNavItem>
-                  <HorizontalNavItem
-                    handleSelectNavItem={handleSelectNavItem}
-                    itemName="money"
-                    isActive={activeNavItem === 'money'}
-                  >
-                    Money
-                  </HorizontalNavItem>
-                  <HorizontalNavItem
-                    handleSelectNavItem={handleSelectNavItem}
-                    itemName="health"
-                    isActive={activeNavItem === 'health'}
-                  >
-                    Health
-                  </HorizontalNavItem>
+                  {!hasReadOnlyPermissionOnly(user) ? (
+                    <HorizontalNavItem
+                      handleSelectNavItem={handleSelectNavItem}
+                      itemName="money"
+                      isActive={activeNavItem === 'money'}
+                      dataTestId="test-applicant-money-section-navigation"
+                    >
+                      Money
+                    </HorizontalNavItem>
+                  ) : (
+                    // eslint-disable-next-line react/jsx-no-useless-fragment
+                    <></>
+                  )}
```

Comment:
This is a valid linting message - it shouldn't be disabled.
The whole expression can be rewritten to avoid useless fragment:

```ts
!hasReadOnlyPermissionOnly(user) && (
   <HorizontalNavItem
      handleSelectNavItem={handleSelectNavItem}
      itemName="money"
      isActive={activeNavItem === 'money'}
      dataTestId="test-applicant-money-section-navigation"
    >
    Money
    </HorizontalNavItem>
)
```

Reply:
I don't seem to be able to use that expression wihout wrapping it in to a fragment which results in the same error. What am I missing?

Reply:
Sorry should have included the update. This is what I tried:
```
{!hasReadOnlyPermissionOnly(user) && (
  <HorizontalNavItem
    handleSelectNavItem={handleSelectNavItem}
    itemName="money"
    isActive={activeNavItem === 'money'}
    dataTestId="test-applicant-money-section-navigation"
  >
    Money
  </HorizontalNavItem>
)}
```
That results in `Type 'false | Element' is not assignable to type 'Element'.  Type 'boolean' is not assignable to type 'ReactElement<any, any>'.ts(2322)`. That expression is widely used though, so not sure what's wrong with it in this case.

Reply:
If I wrap all these child `HorizontalNavItem` components in a fragment then it works. Perhaps that's the way to go. Like this https://github.com/LBHackney-IT/lbh-housing-register/pull/403/commits/9607ca85e49f57438da87256a76516d7eedcca8d

Reply:
Ehh.. the problem is typescript complaining that the return type of the expression is variable and depends on how the condition evaluates. IF false, the returns bool, if true the returns JSX.

Seems like the 2 common solutions are:
1. Do the ternary like you did, except return `null` instead of fragment.
2. Wrap the whole expression into fragment to force an unconditional JSX return type.

Reply:
> If I wrap all these child `HorizontalNavItem` components in a fragment then it works. Perhaps that's the way to go. Like this [9607ca8](https://github.com/LBHackney-IT/lbh-housing-register/commit/9607ca85e49f57438da87256a76516d7eedcca8d)

Wouldn't have guessed that this would work - quite unintuitive outcome given the nature of the error. But if it works, then yeah, let's go with it.

Reply:
Found your problem

https://github.com/LBHackney-IT/lbh-housing-register/blob/4358cf5ad4ddaef374efa287e5c36d1a437f9b41/components/admin/HorizontalNav.tsx#L5

Reply:
Yeah I tried both of those options:
1. Results in the same TS error
2. Results in the useless fragment linting error

Interesting, but the "wrapper" fragments results in seemengly valid TS and linter is OK too and the feature works, so I'll go with that for now.

Reply:
Now it all makes sense. By wrapping the whole thing into fragment, you made it so your `children` that you pass into horrizontalnav is a single JSX element. Where as before it was an array of JSX elements, where some of the elements were `boolean | JSX.Element[]`.

Reply:
At this point the solution is subjective. So I'll give another alternative to choose from:

You could tweak the type linked above to be:
```ts
children: JSX.Element | (JSX.Element | boolean)[];
```

Reply:
This will make it explicit within the parent component's type declaration that some child components are rendered conditionally.

Reply:
That way you should be able to drop the fragment wrapper.

Reply:
Thank you! That's a lot tidier and makes it's more clear how you are supposed to handle the children. Updated heer https://github.com/LBHackney-IT/lbh-housing-register/pull/403/commits/f526803f3b3d9a1eebe4c3195423fd5668f2867c